### ISSUE TITLE
Add ability to specify files to ignore

### ISSUE THREAD

**Issue by dotLou** [markdownlint/markdownlint#167]
Currently`markdownlint` will look for all markdown files known to git. I've got an instance where one of the detected files is one generated by an external tool (https://github.com/skywinder/github-changelog-generator) and is checked into source control. This file has a lot of issues but I don't want it to flag on them because there's nothing I can realistically do to deal with them as everytime I use the external tool, the file will get re-written with the errors.

**Comment by DavidAnson** [comment]
In case it helps, here’s how this change looks in the Node.js implementation: https://github.com/igorshubovych/markdownlint-cli/commit/2640160aae2e285bc92074aa6893f192a278f7fd

**Comment by jamesquilty** [comment]
Another use case: GitLab Issue template files are Markdown and stored in the Project's git repository at `.gitlab/issue_templates/` and violate multiple rules that need to be applied to Markdown files in the rest of the repository. The ability to exclude this subdirectory is important if `mdl` is to be used in [pre-commit](https://pre-commit.com) or a pipeline with the `--git-recurse` flag.

**Comment by jaymzh** [comment]
@dbader - you **can** tell `mdl` to only run on a subdirectory - it takes a directory or a file as an argument. This also works with `-g`, btw.

I agree an exclude option would be useful, probably ala `rubocop`s include and exclude patterns.

**Comment by AlexWayfer** [comment]
It'd be great to exclude files per rules. So, if I have a `README.md` with some HTML stylization — I want to disable corresponding rules only for this file.

**Comment by xorima** [comment]
Hey,

I am the creator of https://github.com/actionshub/markdownlint, this github action now supports a regex to ignore files. I hope this can be of help to you.

Thanks

Jason

**Comment by jamesquilty** [comment]
An in-document ignore tag would be a nice-to-have when/if this feature is implemented. Inline ignore tags are the subject of #16.

**Comment by sylmarch** [comment]
Maybe you have to replace [this part of code](https://github.com/markdownlint/markdownlint/blob/main/lib/mdl.rb#L62-L76)  that is invoking `git ls-files '*.md' '*.markdown'` with a `find` command?
 
 We are currently using this trick in our Gitlab CI/CD to exclude files and directories.

Relevant parts in our `.gitlab-ci.yml` file:

```yml
# [...]

variables:
  # ----------------------------------------------------------------------------
  # LINTER - MDL - CONFIGURATION
  # ----------------------------------------------------------------------------
  # Folders and files not to analyze. Separated with spaces.
  # WARNING: folders and files containing spaces are NOT SUPPORTED.
  MDL_EXCLUDED_FOLDERS: ".git .venv __pycache__ dependencies"
  MDL_EXCLUDED_FILENAMES: ""

# [...]

################################################################################
# LINTERS
################################################################################
lint-markdown:
  stage: lint
  #[...]
  before_script:
    - mdl --version
  script:
    # Compute "find" command options to exclude folders/files
    - |
      find_opts=()

      for FOLDER in ${MDL_EXCLUDED_FOLDERS}; do
          echo "- Excluded folder: : ${FOLDER}"
          find_opts+=('!' '-path' "*/${FOLDER}/*")
      done

      for FILENAME in ${MDL_EXCLUDED_FILENAMES}; do
          echo "- Excluded filename: ${FILENAME}"

          find_opts+=('!' '-name' "${FILENAME}")
      done

    # Find files by extension and invoke "mdl" command on each one
    - find . -type f ${find_opts[@]} \( -name '*.md' -o -name '*.markdown' \) | xargs -r mdl
```

**Comment by dantemogrim** [comment]
I happened to stumble upon exactly what I was looking for within the [VSCode extension's details section](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) interestingly enough.🤷‍♂️ Don't know why this information isn't as available here.

Add a `.markdownlint.json` file in the root of the project with settings you want to change:
```json
{
	"default": true,
	"MD029": false,
	"MD007": { "indent": 2 },
	"no-hard-tabs": false
}
```

Seems to be working.🥳 There's also seems to be support for a `.markdownlint.ignore` file if you want something that works similarly to `.gitignore`. Lots of options available, see the link for details.

It also brings up how you're able to change the settings from a global scope in your `settings.json`.

**Comment by xorima** [comment]
@dantemogrim is this for the ruby markdownlint tool or the nodejs one? https://github.com/DavidAnson/markdownlint

**Comment by DavidAnson** [comment]
Right, the VS Code markdownlint extension uses the Node implementation of markdownlint, not this one.

**Comment by dantemogrim** [comment]
Ah, sorry! My bad!

**Comment by HariSekhon** [comment]
Need this to work around this issue https://github.com/markdownlint/markdownlint/issues/176#issuecomment-1974987909
